src/main/java/com/hackclub/hccore/advancements/NoSleepAdv.java

```diff
@@ -0,0 +1,32 @@
+package com.hackclub.hccore.advancements;
+
+import com.fren_gor.ultimateAdvancementAPI.advancement.Advancement;
+import com.fren_gor.ultimateAdvancementAPI.advancement.BaseAdvancement;
+import com.fren_gor.ultimateAdvancementAPI.advancement.display.AdvancementDisplay;
+import com.fren_gor.ultimateAdvancementAPI.advancement.display.AdvancementDisplayBuilder;
+import com.fren_gor.ultimateAdvancementAPI.util.AdvancementKey;
+import com.fren_gor.ultimateAdvancementAPI.util.CoordAdapter;
+import org.bukkit.Material;
+import org.bukkit.Statistic;
+import org.bukkit.event.player.PlayerBedEnterEvent;
+
+public class NoSleepAdv extends BaseAdvancement {
+  static final AdvancementDisplayBuilder<AdvancementDisplay.Builder, AdvancementDisplay> displayBuilder =
+      new AdvancementDisplay.Builder(Material.RED_BED, "No Sleep???")
+          .challengeFrame()
+          .announceChat()
+          .showToast()
+          .description("Stay awake for one full week");
+  static final int maxProgression = 1;
+
+  public NoSleepAdv(Advancement root, AdvancementKey key,
+      CoordAdapter adapter) {
+    super(key.getKey(), displayBuilder.coords(adapter, key).build(), root, maxProgression);
+
+    registerEvent(PlayerBedEnterEvent.class, e -> {
```

Comment:
Is checking on bed enter the best way to do this?
If so, the description should more accurately reflect the trigger
E.g. go to bed after staying up for a week.